src/coreclr/src/vm/proftoeeinterfaceimpl.cpp

```diff
@@ -1091,7 +1091,7 @@ bool HeapWalkHelper(Object * pBO, void * pvContext)
     OBJECTREF *   arrObjRef      = NULL;
     size_t        cNumRefs       = 0;
     bool          bOnStack       = false;
-    MethodTable * pMT            = pBO->GetMethodTable();
+    MethodTable * pMT            = pBO->GetGCSafeMethodTable();
```

Comment:
>GetGCSafeMethodTable [](start = 40, length = 20)

shouldn't this also be `GetGCSafeTypeHandleIfPossible`?

Reply:
This case is interesting and I am not sure what is the right thing to do here. We cannot simply skip the execution of this method because it requires a return value. Also, right after the call, the code handles collectible right away.